store/displayPreferencesApi.ts

```diff
@@ -0,0 +1,216 @@
+import { DisplayPreferencesDto } from '@jellyfin/client-axios';
+import { MutationTree, ActionTree } from 'vuex';
+import destr from 'destr';
+import {
+  defaultState as settingState,
+  CustomPreferences
+} from '~/store/settings';
+/**
+ * DisplayPreferencesDto but with our custom typings for CustomPrefs property of DisplayPreferencesDto
+ */
+export interface ClientPreferences
+  extends Omit<DisplayPreferencesDto, 'CustomPrefs'> {
+  CustomPrefs?: CustomPreferences;
+}
+
+export interface DisplayPreferencesApiState {
+  syncing: boolean;
+  LastSync: number;
+  LastSettingChange: number;
+}
+
+const defaultState = (): DisplayPreferencesApiState => ({
+  syncing: false,
+  LastSync: -1,
+  LastSettingChange: -1
+});
+
+/**
+ * Cast custom preferences returned from the server from strings to the correct Javascript type
+ *
+ * @param {DisplayPreferencesDto} data - Response from the server
+ * @returns {ClientPreferences} - The response with the correct datatypes.
+ */
+function castResponse(data: DisplayPreferencesDto): ClientPreferences {
+  const result = data as ClientPreferences;
+  for (const [key, value] of Object.entries(
+    result.CustomPrefs as CustomPreferences
+  )) {
+    /**
+     * destr does proper conversion for all the types, even undefined
+     */
+    // @ts-expect-error - TypeScript can't infer types from Object.entries
+    result.CustomPrefs?.[key] = destr(value);
+  }
+  return result;
+}
+
+export const state = defaultState;
+
+export const mutations: MutationTree<DisplayPreferencesApiState> = {
+  RESET_STATE(state: DisplayPreferencesApiState) {
+    Object.assign(state, defaultState);
+  },
+  SYNCING_STARTED(state: DisplayPreferencesApiState) {
+    state.syncing = true;
+  },
+  SYNCING_ENDED(state: DisplayPreferencesApiState) {
+    state.syncing = false;
+    state.LastSync = Date.now();
+  },
+  UPDATE_CLIENT_SETTINGS(state: DisplayPreferencesApiState) {
+    state.LastSettingChange = Date.now();
+  }
+};
+
+export const actions: ActionTree<
+  DisplayPreferencesApiState,
+  DisplayPreferencesApiState
+> = {
+  /**
+   * Fetches display preferences and stores them at boot time
+   *
+   * @param {any} context - Vuex action context
+   * @param {any} context.dispatch - Vuex dispatch
+   */
+  async initState({ commit, dispatch }) {
```

Comment:
Should probably be `fetchPreferences` or something. `initState` seems like something you do once, but this fetches from the server in case the local store is stale, which means it can happen even when the state it already initialized (For example, the user goes in offline mode for a bit, changes a setting on another instance of the client, connects back up and remote settings are stale. When the websocket gets back online, it'd need to run this to sync back up).